test/parallel/test-stream-readable-unshift.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { Readable } = require('stream');
+
+{
+  // Check that strings are saved as Buffer
+  const readable = new Readable({ read() {} });
+
+  const string = 'abc';
+  readable.unshift(string);
+
+  assert.strictEqual(
+    Buffer.isBuffer(readable._readableState.buffer.head.data),
+    true
+  );
+
+  // Check default encoding is utf8
+  assert.strictEqual(
+    readable._readableState.buffer.head.data.toString('utf8'),
+    string
+  );
+
+}
+
+{
+  // Check that data goes at the beginning
+  const readable = new Readable({ read() {} });
+  const string = 'front';
+
+  readable.push('b');
+  readable.unshift(string);
+
+  assert.strictEqual(
+    readable._readableState.buffer.head.data.toString('utf8'),
+    string
+  );
```

Comment:
This can be tested by consuming the stream and verifying that the data comes out in order.